nemo/collections/nlp/modules/common/transformer/transformer_decoders.py

```diff
@@ -138,6 +138,10 @@ def __init__(
             pre_ln,
         )
         self.layers = nn.ModuleList([copy.deepcopy(layer) for _ in range(num_layers)])
+
+        if pre_ln:
+            self.layers.append(nn.LayerNorm(hidden_size, eps=1e-5))
```

Comment:
Yeah this seems like a good idea to avoid adding if statements in the loop below. You can rename `self.layers` to `self.transformer_layers`. Also for backward compatibility for models without this final layernorm, can you please add a `pre_ln_final_form=False` default to the encoder/decoder class? Does that seem like a reasonable solution? @okuchaiev.

Reply:
Changing the name from self.layers to self.transformer_layers may break older checkpoints?

Reply:
Ah yes, my bad.

Reply:
> is there a reason eps is set here?

@VahidooX  Authors didn't mention eps in final norm in their paper. Here just follow all the layer norms in blocks.

Thanks for the suggestions. Updated!

Reply:
It is fine as the default of Pytorch is also 1e-5 and they would never change such defaults. You may keep it.